packages/react/src/components/Sheet/use-smooth-scroll.tsx

```diff
@@ -138,19 +200,54 @@ export const useSmoothScroll = (
         ) {
           return; // still a tap – do nothing
         }
+
+        // Determine primary scroll direction based on initial movement
```

Comment:
Direction locking- as earlier if scroll direction from user scroll is not determine it use to scroll a bit in both direction. Now it locks in which direction it have to scroll.